### PR TITLE
Support unit testing Perform funcs which push jobs

### PR DESCRIPTION
**Pull request by mperham**
See #51 for details. WDYT?

### CODE REVIEW

**Comment by bploetz** [comment]
@mperham I'll have to pull this down and try it for real, but this looks like it solves the problem nicely! 👍

**Comment by mperham** [comment]
If you can think of a clean way to trigger Push() errors, that might be a nice addition to this better testing support.

**Comment by bploetz** [comment]
@mperham could you simply add an additional argument to the `ExecuteWithResult` function:

```
func (tp *testExecutor) ExecuteWithResult(specjob *faktory.Job, p Perform, simulatePushError bool) (ExecuteResult, error) 
```

Then pass the `simulatePushError` flag down to the internal methods accordingly, and if it's set to true have `Push` return an error.

**Comment by bploetz** [comment]
@mperham do you have any plans to release something like to support testing jobs that enqueue other jobs? Since [you suggested using `helper.With()` to push jobs within jobs](https://github.com/contribsys/faktory/issues/373#issuecomment-1242805855), I'm now back to [this original problem](https://github.com/contribsys/faktory_worker_go/issues/51) where I can't test doing it that way.

**Comment by mperham** [comment]
I add features based on demand these days. If no one asks for it, I don't add it. Send a PR if you have an initial spike we can collaborate on.

**Comment by bploetz** [comment]
Isn't this PR an initial spike?

**Comment by mperham** [comment]
Sorry, didn’t notice the context, just replying on my phone.

File: context.go [hunk 1]
```diff
@@ -68,6 +71,47 @@ type Helper interface {
...
+		return b.(string)
+	}
+	return ""
+}
+func (h *testHelper) JobType() string {
+	return h.job.Type
+}
+
+func (h *testHelper) TrackProgress(percent int, desc string, reserveUntil *time.Time) error {
+	return nil
```
Comment (mperham) [inline context.go:96]: Instead of creating a separate type which doesn't implement most of the interface, maybe use a swappable Push() fn?

**Comment by mperham** [comment]
Wait, I created this PR??? 😆 I'm very confused.

**Comment by mperham** [comment]
Unfortunately I'm not going to be able to get to this for a while. If you want to create a new PR which solves your issues, I'd be happy to review.

**Pull request closed without merging**